decapodes.it

```diff
@@ -0,0 +1,137 @@
+@sum Parameter begin
+  LiteralParameter(type::String, value::Float64)
+  ConstantParameter(type::String, value::String)
+end
+
+struct Condition
+  type::String
```

Comment:
Should this be an enumerated list via a sum type? What are the options here?

```julia
@sum ConditionType begin
  Initial
  Boundary
  Interface
end

struct Condition
  type::ConditionType
  value::?
  domain_mesh::?
end
```

Reply:
It's unclear what the conditions type should be but I don't think it can be replace by what you've stated above.

It would be nice in the future to combine all the conditions into one list but I don't want to change the spec @liunelson and @mattprintz set up.